src/Module/Contacts.php

```diff
@@ -0,0 +1,1137 @@
+				$last_update .= ' ' . (($contact['last-update'] <= $contact['success_update']) ? L10n::t("\x28Update was successful\x29") : L10n::t("\x28Update was not successful\x29"));
+			}
+			$lblsuggest = (($contact['network'] === Protocol::DFRN) ? L10n::t('Suggest friends') : '');
+
+			$poll_enabled = in_array($contact['network'], [Protocol::DFRN, Protocol::OSTATUS, Protocol::FEED, Protocol::MAIL]);
+
+			$nettype = L10n::t('Network type: %s', ContactSelector::networkToName($contact['network'], $contact["url"]));
+
+			// tabs
+			$tab_str = self::contacts_tab($a, $contact, 3);
+
+			$lost_contact = (($contact['archive'] && $contact['term-date'] > NULL_DATE && $contact['term-date'] < DateTimeFormat::utcNow()) ? L10n::t('Communications lost with this contact!') : '');
+
+			$fetch_further_information = null;
+			if ($contact['network'] == Protocol::FEED) {
+				$fetch_further_information = [
+					'fetch_further_information',
+					L10n::t('Fetch further information for feeds'),
+					$contact['fetch_further_information'],
+					L10n::t("Fetch information like preview pictures, title and teaser from the feed item. You can activate this if the feed doesn't contain much text. Keywords are taken from the meta header in the feed item and are posted as hash tags."),
+					['0' => L10n::t('Disabled'),
+						'1' => L10n::t('Fetch information'),
+						'3' => L10n::t('Fetch keywords'),
+						'2' => L10n::t('Fetch information and keywords')
+					]
+				];
+			}
+
+			$poll_interval = null;
+			if (in_array($contact['network'], [Protocol::FEED, Protocol::MAIL])) {
+				$poll_interval = ContactSelector::pollInterval($contact['priority'], (!$poll_enabled));
+			}
+
+			$profile_select = null;
+			if ($contact['network'] == Protocol::DFRN) {
+				$profile_select = ContactSelector::profileAssign($contact['profile-id'], (($contact['network'] !== Protocol::DFRN) ? true : false));
+			}
+
+			/// @todo Only show the following link with DFRN when the remote version supports it
+			$follow = '';
+			$follow_text = '';
+			if (in_array($contact['rel'], [Contact::FRIEND, Contact::SHARING])) {
+				if (in_array($contact['network'], Protocol::NATIVE_SUPPORT)) {
+					$follow = System::baseUrl(true) . "/unfollow?url=" . urlencode($contact["url"]);
+					$follow_text = L10n::t("Disconnect/Unfollow");
+				}
+			} else {
+				$follow = System::baseUrl(true) . "/follow?url=" . urlencode($contact["url"]);
+				$follow_text = L10n::t("Connect/Follow");
+			}
+
+			// Load contactact related actions like hide, suggest, delete and others
+			$contact_actions = self::contact_actions($contact);
+
+			if ($contact['uid'] != 0) {
+				$lbl_vis1 = L10n::t('Profile Visibility');
+				$lbl_info1 = L10n::t('Contact Information / Notes');
+				$contact_settings_label = L10n::t('Contact Settings');
+			} else {
+				$lbl_vis1 = null;
+				$lbl_info1 = null;
+				$contact_settings_label = null;
+			}
+
+			$tpl = get_markup_template("contact_edit.tpl");
+			$o .= replace_macros($tpl, [
```

Comment:
Nope, but that's the beauty of arrays, isn't it @annando ? 😋

Reply:
I think we really can be sure that the array values do exist there. We don't need to check for every special case. And even when we would create some result object from database queries then we could have the situation where the selected fields don't contain all fields that we later want to use. So there would be some notice as well. I don't see a difference there.

Reply:
The difference is that with a `Contact` row object you always know what data to expect because the object has explicit properties. If you add a field to the database, you still know what properties your object has, which prevent Undefined notices. And if you remove a field from the database, you won't get just a Notice and unexpected behavior but a proper error from the Contact row object.

In the end, the goal is to close as many blind spot as you can, even if arrays would work 95% of the time.